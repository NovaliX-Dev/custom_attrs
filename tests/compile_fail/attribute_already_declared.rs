use custom_attrs::CustomAttrs;

#[derive(CustomAttrs)]
#[attr(pub a: usize)]
#[attr(pub a: usize)]
enum Enum {
    #[attr(a = 3)]
    Variant1,

    #[attr(a = 3)]
    Variant2
}

#[derive(CustomAttrs)]
#[attr(
    pub b: Option<usize>,
    pub b: Option<usize>,
)]
enum Enum2 {
    Variant1,

    #[attr(a = 3)]
    Variant2
}

fn main() {
    // let _a = Enum::Variant1.get_a();
}