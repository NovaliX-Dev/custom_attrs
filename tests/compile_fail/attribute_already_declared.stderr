error: This attribute is already declared.
 --> tests/compile_fail/attribute_already_declared.rs:5:12
  |
5 | #[attr(pub a: usize)]
  |            ^

error: `a` is already declared here.
 --> tests/compile_fail/attribute_already_declared.rs:4:12
  |
4 | #[attr(pub a: usize)]
  |            ^

error: This attribute is already declared.
  --> tests/compile_fail/attribute_already_declared.rs:17:9
   |
17 |     pub b: Option<usize>,
   |         ^

error: `b` is already declared here.
  --> tests/compile_fail/attribute_already_declared.rs:16:9
   |
16 |     pub b: Option<usize>,
   |         ^
